### PR TITLE
fix: columns use client side property for identification

### PR DESCRIPTION
**Pull request by LiKang6688**

### CODE REVIEW

File: src/ext.js [hunk 1]
```diff
@@ -293,11 +288,6 @@ const definition = {
               defaultValue: true,
               show: false,
             },
-            cId: {
```
Comment (haxxmaxx) [inline src/ext.js:296]: Above here, autosort is still included, so it will be declared twice

File: src/table/handle-data.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ export function getColumnInfo(layout, colIndex) {
       isDim,
       width: 200,
       label: info.qFallbackTitle,
-      id: info.cId,
+      id: index,
```
Comment (haxxmaxx) [inline src/table/handle-data.js:23]: `colIndex` should work fine. All the key needs to be is a unique identifier in the current list. Not sure if necessarry, but I think a prefix would be nice, so `col-0` etc

File: src/table/handle-data.js [hunk 3]
```diff
@@ -36,7 +36,7 @@ export default async function manageData(model, layout, pageInfo) {
   const matrix = dataPages[0].qMatrix;
 
   // using filter to remove hidden columns (represented with false)
-  const columns = columnOrder.map((c) => getColumnInfo(layout, c)).filter(Boolean);
+  const columns = columnOrder.map((c, index) => getColumnInfo(layout, c, index)).filter(Boolean);
   const rows = matrix.map((r, rowIdx) => {
     const row = { key: rowIdx };
```
Comment (haxxmaxx) [inline src/table/handle-data.js:41]: Same thing here, `row-0`...

Comment (haxxmaxx) [inline src/table/handle-data.js:41]: Same thing here, `row-0`...

File: src/table/handle-data.js (hunk 2 above)
Comment (LiKang6688) [reply src/table/handle-data.js:24]: sure

File: src/ext.js (hunk 1 above)
Comment (LiKang6688) [reply src/ext.js:296]: Deleted.

File: src/table/handle-data.js [hunk 4]
```diff
@@ -36,9 +36,9 @@ export default async function manageData(model, layout, pageInfo) {
   const matrix = dataPages[0].qMatrix;
 
   // using filter to remove hidden columns (represented with false)
-  const columns = columnOrder.map((c) => getColumnInfo(layout, c)).filter(Boolean);
+  const columns = columnOrder.map((c, index) => getColumnInfo(layout, c, index)).filter(Boolean);
   const rows = matrix.map((r, rowIdx) => {
-    const row = { key: rowIdx };
+    const row = { key: `col-${rowIdx}` };
```
Comment (haxxmaxx) [inline src/table/handle-data.js:41]:
```suggestion
    const row = { key: `row-${rowIdx}` };
```

File: src/table/handle-data.js [hunk 5]
```diff
@@ -20,7 +20,7 @@ export function getColumnInfo(layout, colIndex) {
       isDim,
       width: 200,
       label: info.qFallbackTitle,
-      id: info.cId,
+      id: `col-${index}`,
```
Comment (haxxmaxx) [inline src/table/handle-data.js:23]: I meant that you should use `colIndex` instead of `index`

**Review by haxxmaxx — APPROVED** [review]

**Pull request merged**